integreat_compass/cms/views/interactions/vote_form_view.py

```diff
@@ -0,0 +1,87 @@
+from django.db.models import Case, Count, When
+from django.shortcuts import render
+from django.views.generic import TemplateView
+
+from ...constants import offer_version_states
+from ...models import OfferVersion, User
+
+
+class VoteFormView(TemplateView):
+    """
+    View for the vote form
+    """
+
+    template_name = "interactions/vote_form.html"
+
+    @staticmethod
+    def get_pending_offer_versions(request, number_of_board_members):
+        """
+        Get all offer versions that need a vote from currently logged-in user
+
+        :param request: Object representing the user call
+        :type request: ~django.http.HttpRequest
+
+        :param number_of_board_members: current number of board member
+        :type number_of_board_members: int
+
+        :return: list of offer versions with all details
+        """
+        latest_version_of_offer = OfferVersion.objects.order_by(
+            "offer", "-created_at"
+        ).distinct("offer")
+        offer_versions = (
+            OfferVersion.objects.prefetch_related("votes")
+            .select_related("offer")
+            .select_related("offer__organization")
+            .prefetch_related("offer__tags")
+            .annotate(approves=Count(Case(When(votes__approval=True, then=1))))
+            .annotate(declines=Count(Case(When(votes__approval=False, then=1))))
+            .annotate(pending=number_of_board_members - Count("votes"))
+            .annotate(user_voted_on=Case(When(votes__creator=request.user, then=True)))
```

Comment:
```suggestion
            .annotate(
                approves=Count("votes", filter=Q(votes__approval=True)),
                declines=Count("votes", filter=Q(votes__approval=False)),
                pending=number_of_board_members - Count("votes"),
                user_voted_on=Count("votes", filter=Q(votes__creator=request.user)),
            )
```
The [docs](https://docs.djangoproject.com/en/4.2/topics/db/aggregation/#filtering-on-annotations) suggest a slightly different syntax, but it doesn't make a huge difference.